python/lsst/summit/extras/plotting/psfPlotting.py

```diff
@@ -0,0 +1,524 @@
+# This file is part of summit_extras.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = [
+    "addColorbarToAxes",
+    "makeTableFromSourceCatalogs",
+    "makeFigureAndAxes",
+    "extendTable",
+    "makeFocalPlanePlot",
+    "makeEquatorialPlot",
+    "makeAzElPlot",
+]
+
+
+from typing import TYPE_CHECKING
+
+import matplotlib.pyplot as plt
+import numpy as np
+from astropy.table import vstack
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+from lsst.afw.cameraGeom import FOCAL_PLANE
+from lsst.afw.geom.ellipses import Quadrupole
+from lsst.geom import LinearTransform, radians
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from astropy.table import Table  # noqa: F401
+    from matplotlib.colorbar import Colorbar  # noqa: F401
+
+    from lsst.afw.cameraGeom import Camera  # noqa: F401
+    from lsst.afw.image import VisitInfo  # noqa: F401
+    from lsst.afw.table import SourceCatalog  # noqa: F401
```

Comment:
I don't think I've ever seen complaints about cases like this (i.e type annotation use only) from Ruff and Flake8, and those linters _do_ flag cases where symbols are not used in either runtime _or_ type annotations.  Maybe there's a configuration option that should be toggled?  Or are you using some other linter?

Reply:
Huh, no, it's probably fine, of from before it was put inside an `if TYPE_CHECKING:` block or something like that, thanks.